### PR TITLE
fix(PE-6486): Capture errors from arns emitter

### PR DESCRIPTION
**Pull request by atticusofsparta**

### CODE REVIEW

File: src/hooks/useWalletANTs/useWalletANTs.tsx [hunk 1]
```diff
@@ -440,6 +442,7 @@ export function useWalletANTs() {
         dispatch: dispatchArNSState,
         emitter: arnsEmitter,
         walletAddress,
+        registryId: arnsContractId,
```
Comment (kunstmusik) [inline src/hooks/useWalletANTs/useWalletANTs.tsx:445]: I think we've got three different names for the same thing now:

* registryId
* arnsContractId
* ioProcessId

If these are all the same thing, we should settle one a single name.

File: src/state/actions/dispatchArNSUpdate.ts [hunk 2]
```diff
@@ -40,6 +44,37 @@ export function dispatchArNSUpdate({
       payload: totalIds,
     });
   });
+  const errorHandler = (e: string) => {
+    if (e.startsWith('Error getting ArNS records')) {
+      eventEmitter.emit(
+        'error',
+        new Error('Unable to load ArNS records. Please refresh to try again.'),
+      );
+      // Behaviour note: this will only stop the loading UI, it will still display the
```
Comment (kunstmusik) [inline src/state/actions/dispatchArNSUpdate.ts:53]: This comment just ends with "the", seems like some more is meant to be mentioned?

File: src/state/actions/dispatchArNSUpdate.ts [hunk 3]
```diff
@@ -40,6 +44,37 @@ export function dispatchArNSUpdate({
...
+        tags: {
+          walletAddress: walletAddress.toString(),
+          registryId: registryId.toString(),
+        },
+      });
+    } else if (!e.includes('does not support provided action.')) {
+      eventEmitter.emit('error', new Error(e));
+    }
+  };
+  // error listnener handles timeout
```
Comment (kunstmusik) [inline src/state/actions/dispatchArNSUpdate.ts:74]: typo: listener

File: src/state/actions/dispatchArNSUpdate.ts [hunk 4]
```diff
@@ -51,5 +86,9 @@ export function dispatchArNSUpdate({
     });
   });
 
-  emitter.fetchProcessesOwnedByWallet({ address: walletAddress.toString() });
+  emitter
+    .fetchProcessesOwnedByWallet({ address: walletAddress.toString() })
+    .catch((e) =>
+      errorHandler('Error getting processes owned by wallet: ' + e.message),
```
Comment (kunstmusik) [inline src/state/actions/dispatchArNSUpdate.ts:92]: Would it be clearer to the user to mention "ANT processes"?

File: src/state/actions/dispatchArNSUpdate.ts (hunk 2 above)
Comment (atticusofsparta) [reply src/state/actions/dispatchArNSUpdate.ts:53]: Realized the comment was incorrect anyways, removing.

File: src/state/actions/dispatchArNSUpdate.ts (hunk 4 above)
Comment (atticusofsparta) [reply src/state/actions/dispatchArNSUpdate.ts:92]: Specified assets instead sinces both domains and ants are affected

File: src/state/actions/dispatchArIOContract.ts [hunk 5]
```diff
@@ -1,17 +1,24 @@
...
   contract: AoIORead | AoIOWrite;
+  ioProcessId: ArweaveTransactionID;
   dispatch: Dispatch<GlobalAction>;
 }) {
   dispatch({
     type: 'setArIOContract',
     payload: contract,
   });
+  dispatch({
+    type: 'setioProcessId',
```
Comment (kunstmusik) [inline src/state/actions/dispatchArIOContract.ts:21]: looks strange, suggest to change this to "setIoProcessId" and in reducer

Comment (atticusofsparta) [reply src/state/actions/dispatchArIOContract.ts]: fixed

**Review by kunstmusik — APPROVED** [review]

**Pull request merged**